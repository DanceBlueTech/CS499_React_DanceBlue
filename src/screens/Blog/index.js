// Import third-party dependencies
import React from "react";
import { Text, View, StyleSheet } from "react-native";
import { WebView } from "react-native-webview";

// Component for profile screen in main navigation
export class BlogScreen extends React.Component {
  render() {
    return (
      <View style={{ flex: 1}}>
      <WebView source={{ uri: "https://www.danceblue.org/blog/" }} />
      </View>
    );
  }
}
