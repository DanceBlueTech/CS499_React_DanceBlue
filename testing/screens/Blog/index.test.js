import React from 'react';
import renderer from 'react-test-renderer';

import { BlogScreen } from '../../../src/screens/Blog/index';

it('renders Blog page correctly', () => {
    expect(renderer.create(<BlogScreen/>)).toMatchSnapshot();
});
// to run test: npm run test