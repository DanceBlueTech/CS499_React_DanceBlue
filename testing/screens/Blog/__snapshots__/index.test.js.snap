// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Blog page correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "https://www.danceblue.org/blog/",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "borderWidth": 0,
            "flex": 1,
            "width": 750,
          },
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
